cypress-io/cypress
# feat: pass parser options to allow parsing of tsx files

**NOTE**: All of our cypress-in-cypress tests will fail on this branch because of an issue with the way that Babel dependencies are being resolved in the v8 snapshot. This is currently being investigated. I'm going to merge this into the feature branch anyway once I get approvals and I see that `CodegenActions.spec.ts` has passed in CI.

### User facing changelog


n/a, merging to feature branch

### Additional details


Our current implementation of create from React component will fail to parse TSX files that contain type annotations. This PR adds some parser plugins to allow for the parsing of TSX files.

### Steps to test


1. Check out this branch
2. Revert the change to `CodegenActions.ts`
3. Run `env DEBUG="cypress:data-context" yarn workspace @packages/data-context test`
4. Notice that the `correctly parses typescript files` test case fails
5. See the debug error in the console indicating a syntax error occurred while parsing the file
6. Re-introduce the changes to `CodegenActions.ts`
7. Run the tests again and verify that they all pass and that `LoginForm.tsx` was parsed correctly

### How has the user experience changed?


n/a

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
